website/content/docs/platform/k8s/vso/sources/vault/client-cache.mdx

```diff
@@ -0,0 +1,266 @@
+---
+layout: docs
+page_title: Enable the encrypted Vault client cache
+description: >-
+  Learn how to enable the operator's encrypted Vault Client cache for dynamic secrets
+---
+
+# Enable the encrypted Vault client cache
+
+The Vault Secrets Operator supports storing and encrypting its Vault client cache using Vault Server's [transit secrets engine](/vault/docs/secrets/transit). By default client cache storage is not enabled, since it requires extra Vault Server setup, but this guide will walk you through the steps to enable and configure it.
+
+<Highlight>
+
+Enabling encrypted client cache storage is recommended especially when using [VaultDynamicSecrets](/vault/docs/platform/k8s/vso/api-reference#vaultdynamicsecret).
+
+</Highlight>
+
+## Step 1: Configure transit
+
+Use the following commands/terraform to enable the transit secret engine, a key for the client cache, and a policy for the operator to use the key.
+
+<CodeTabs>
+<CodeBlockConfig>
+
+```shell-session
+export VAULT_NAMESPACE=<VAULT_NAMESPACE>
+
+vault secrets enable transit
+
+vault write -f transit/keys/vso-client-cache
+
+vault policy write operator - <<EOH
+path "transit/encrypt/vso-client-cache" {
+  capabilities = ["create", "update"]
+}
+path "transit/decrypt/vso-client-cache" {
+  capabilities = ["create", "update"]
+}
+EOH
+```
+
+</CodeBlockConfig>
+<CodeBlockConfig>
+
+```hcl
+resource "vault_mount" "transit" {
+  namespace   = "<VAULT_NAMESPACE>"
+  path        = "transit"
+  type        = "transit"
+  description = "VSO Client Cache"
+}
+
+resource "vault_transit_secret_cache_config" "cache" {
+  namespace = vault_mount.transit.namespace
+  backend   = vault_mount.transit.path
+  size      = 500
+}
+
+resource "vault_transit_secret_backend_key" "cache" {
+  namespace        = vault_mount.transit.namespace
+  backend          = vault_mount.transit.path
+  name             = "vso-client-cache"
+  deletion_allowed = true
+}
+
+resource "vault_policy" "operator" {
+  namespace = vault_transit_secret_backend_key.cache.namespace
+  name      = "operator"
+  policy    = <<EOT
```

Comment:
nit: I think this could be refactored slightly cleaner with a [`vault_policy_document`](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/data-sources/policy_document) data source.

Reply:
Thanks! That is better, done in 1f523bf.